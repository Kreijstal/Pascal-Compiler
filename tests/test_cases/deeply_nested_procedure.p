program deeply_nested_test(output);
var
    global_var: integer;

procedure outer;
var
    outer_local: integer;

    procedure mid;
    var
        mid_local: integer;

        procedure inner;
        var
            inner_local: integer;
        begin
            inner_local := 10;
            outer_local := outer_local + inner_local;
            global_var := global_var + outer_local + mid_local + inner_local;
            writeln(outer_local);
            writeln(global_var);
        end;

    begin
        mid_local := 3;
        inner;
        writeln(outer_local);
    end;

begin
    outer_local := 5;
    mid;
end;

begin
    global_var := 1;
    outer;
    writeln(global_var);
end.
