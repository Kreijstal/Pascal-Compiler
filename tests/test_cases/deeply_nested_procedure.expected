15
29
15
29
